relay-metrics/src/aggregator.rs

```diff
@@ -891,11 +891,11 @@ impl fmt::Debug for Aggregator {
 /// Splits buckets into N logical partitions, determined by the bucket key.
 pub fn partition_buckets(
     project_key: ProjectKey,
-    buckets: Vec<Bucket>,
+    buckets: impl IntoIterator<Item = Bucket>,
     flush_partitions: Option<u64>,
 ) -> BTreeMap<Option<u64>, Vec<Bucket>> {
     let flush_partitions = match flush_partitions {
-        None => return BTreeMap::from([(None, buckets)]),
+        None => return BTreeMap::from([(None, buckets.into_iter().collect())]),
```

Comment:
I checked with [godbolt](https://godbolt.org/z/o41qrWbhj) this is optimized away when passing a vector already